src/Map.jsx

```diff
@@ -98,60 +127,60 @@ class Map extends Component {
       null,
       (err) => {
         if (err) {
-          throw (err);
+          console.warn(err);
         }
       },
       { force: true },
     );
   }
-  /**
-   * return coordinates in an array [lat, lon]
-   */
-  getCoordinateArray(stop) {
-    return [stop.lat, stop.lon];
-  }
   /**
    * given the two selected stop sids, returns a line segment
    * between them
    */
-  getRouteBetweenStops(routeStops) {
-    const stopSids = this.state.selectedStops.map(stop => stop.sid);
+  getRouteBetweenStops(routeStops, stops) {
+    const stopSids = stops.map(stop => stop.sid);
     stopSids.sort((a, b) => a - b);
-    const route = routeStops.map(stop => this.getCoordinateArray(stop));
-    let startingPoint = routeStops.find(stop => stop.sid === stopSids[0]);
-    startingPoint = this.getCoordinateArray(startingPoint);
-    startingPoint = turf.point(startingPoint);
-    let endingPoint = routeStops.find(stop => stop.sid === stopSids[1]);
-    endingPoint = this.getCoordinateArray(endingPoint);
-    endingPoint = turf.point(endingPoint);
+    const route = routeStops.map(stop => [stop.lon, stop.lat]);
+    const startingPointStop = new Stop(routeStops.find(stop => stop.sid === stopSids[0]));
+    const endingPointStop = new Stop(routeStops.find(stop => stop.sid === stopSids[1]));
+    /*
+    * if either value is undefined, it means user selected another stop on another route.
+    * so clear all stops and subroute
+    */
+    if (startingPointStop.isUndefined() || endingPointStop.isUndefined()) {
+      this.setState({ subroute: null, selectedStops: [] });
+      return;
+    }
+    const startingPoint = turf.point(startingPointStop.getCoordinateArray());
+    const endingPoint = turf.point(endingPointStop.getCoordinateArray());
     const line = turf.lineString(route);
-    return turf.lineSlice(startingPoint, endingPoint, line);
+    const subroute = turf.lineSlice(startingPoint, endingPoint, line);
+    this.setState({ subroute, selectedStops: stops });
   }
   /**
    * sets stop sids based on selected stops.
    * Stores up to two stops sids. Used to draw subroutes
    */
   getStopInfo(route, stopCoordinates) {
-    let stops = this.state.selectedStops;
-    const station
-    = route.stops.find(currentStop => currentStop.lon === stopCoordinates[0]
+    let [...stops] = this.state.selectedStops;
```

Comment:
Just curious, what made you change this from `let stops` to `let [...stops]`? Is it to make a copy? In that case, I think `let stops = [...this.state.selectedStops]` is the way to go based on the airbnb styleguide - 
https://github.com/airbnb/javascript#arrays--from-iterable